### DISCUSSION TITLE
Whats the best way to simulate gene transduction in a COBRA model?

### DISCUSSION THREAD

**Discussion by mbutawan** [opencobra/cobrapy#1106] [Q&A] [answered]
I've seen how easy it is to simulate KO experiments with cobrapy, but I have not seen anything really about transduction/introduction of foreign genes to an existing model. Can someone point me in the right direction?

Is it best to just create the metabolites/reactions and not worry about the genes themselves? Is the gpr (gene_reaction_rule) required to simulate metabolic flux?

**Comment by Midnighter** [comment, marked as answer]
Hello @mbutawan,

Introducing new genes is a bit of manual work. If you know exactly what genes you want to introduce and the metabolic reaction catalyzed by the gene product, then I would just define a new reaction with the corresponding metabolites. You don't need to define the GPR in order to simulate flux but it can be nice as annotation/information and, of course, if you wanted to knock out the gene.

If you only know that you want to produce a certain compound, then there are multiple tools available for pathway selection, for example, [cameo](https://cameo.bio/).

**Reply by mbutawan** [reply]
Hello @Midnighter,
Thank you for answering and explaining the necessary procedures. This is very helpful for me as a newcomer to this amazing tool!

Also thank you for the introduction to the new library! I'm sure I'll find a use for that in some project.
